sway-lib-std/src/math.sw

```diff
@@ -118,14 +157,21 @@ impl Power for u64 {
 
 impl Power for u32 {
     fn pow(self, exponent: u32) -> Self {
-        let res = asm(r1: self, r2: exponent, r3) {
+        let mut res = asm(r1: self, r2: exponent, r3) {
             exp r3 r1 r2;
             r3: u64
         };
-        // If panic on wrapping math is enabled, only then revert
-        if flags() & F_WRAPPING_DISABLE_MASK == 0 {
-            assert(res <= Self::max().as_u64());
+
+        if res > Self::max().as_u64() {
+            // If panic on wrapping math is enabled, only then revert
+            if flags() & F_WRAPPING_DISABLE_MASK == 0 {
```

Comment:
Isn't there a `panic_on_overflow_enabled()` function for the std-lib?

Reply:
that pr isnt merged yet, if that gets in before this ill change this